src/libpgmoneta/utils.c

```diff
@@ -1967,17 +1968,27 @@ do_copy_file(struct worker_common* wc)
    char* dn = NULL;
    char* to = NULL;
 
-   fd_from = open(fi->from, O_RDONLY);
+   /* if the file is partial try for complete file */
+   if (!pgmoneta_is_file(fi->from) && pgmoneta_ends_with(fi->from, ".partial"))
```

Comment:
This should be caller's duty, try copying partial, if failed try again with full segment. Copy itself should just faithfully try to copy whatever it's asked to. But it's ok here since you are just patching it to make CI happy. We'll address this in another issue.